stdlib/Logging/src/ConsoleLogger.jl

```diff
@@ -118,8 +118,17 @@ function handle_message(logger::ConsoleLogger, level::LogLevel, message, _module
     end
 
     # Generate a text representation of the message and all key value pairs,
-    # split into lines.
-    msglines = [(indent=0, msg=l) for l in split(chomp(convert(String, string(message))::String), '\n')]
+    # split into lines.  This is specialised to improve type inferesence,
+    # and reduce the risk of resulting method invalidations.
+    message = string(message)
+    msglines = if Base._isannotated(message)
+        message = Base.AnnotatedString(String(message), Base.annotations(message))
+        @NamedTuple{indent::Int, msg::SubString{<:Union{Base.AnnotatedString{String}, String}}}[
```

Comment:
```suggestion
        @NamedTuple{indent::Int, msg::SubString{Base.AnnotatedString{String}}[
```

Reply:
This change would be invalid since later code pushes values like `(indent=2, msg=SubString("$key ="))` to `msglines`.

Reply:
That shouldn't be invalid, if the convert methods are defined correctly.

But anyways, that suggest this should have the same type on both branches then, so you should update the other branch as well to have this type. I think you can slightly simplify it by making it a pair of concrete type also, instead of a UnionAll type.
```suggestion
        @NamedTuple{indent::Int, msg::Union{SubString{Base.AnnotatedString{String}}, SubString{String}}}}[
```

Reply:
Oh? I recall widening this type because of an issue I ran into when developing this PR, but October last year is too far back for me to recall what was happening clearly.

I've applied the concrete-ification you've suggested.